### PR TITLE
Add support for scalaJSOutputPatterns

### LINKED ISSUE #1721: Support Scala.js linker config `OutputPatterns`
In order to be able to run `ESModule`s on Node without tinkering with `package.json`, linker output needs to have `.mjs` extension instead of `.js`. Scala.js linker supports this via [`OutputPatterns`](https://www.scala-js.org/doc/project/module.html#es-modules-and-nodejs). Mill `ScalaJSModule` doesn't expose that config in its API.

Also, would it make sense to expose Scala.js linker API directly instead of the current facade API?

### PR DESCRIPTION
**Pull request by lolgab**
Fixes #1721

### CODE REVIEW

File: scalajslib/src/mill/scalajslib/api/ScalaJSApi.scala [hunk 1]
```diff
@@ -107,3 +107,93 @@ object JsEnvConfig {
       autoExit: Boolean
   ) extends JsEnvConfig
 }
+
+class OutputPatterns private (
+    private[scalajslib] val jsFile: String,
+    private[scalajslib] val sourceMapFile: String,
+    private[scalajslib] val moduleName: String,
+    private[scalajslib] val jsFileURI: String,
+    private[scalajslib] val sourceMapURI: String
```
Comment (lefou) [inline scalajslib/src/mill/scalajslib/api/ScalaJSApi.scala:116]: Why are these private?

File: scalajslib/worker/0.6/src/ScalaJSWorkerImpl.scala [hunk 2]
```diff
@@ -83,7 +83,8 @@ class ScalaJSWorkerImpl extends ScalaJSWorkerApi {
       sourceMap: Boolean, // ignored in 0.6
       moduleKind: ModuleKind,
       esFeatures: ESFeatures,
-      moduleSplitStyle: ModuleSplitStyle // ignored in 0.6
+      moduleSplitStyle: ModuleSplitStyle, // ignored in 0.6
+      outputPatterns: OutputPatterns // ignored in 0.6
```
Comment (lefou) [inline scalajslib/worker/0.6/src/ScalaJSWorkerImpl.scala:87]: It makes sense to focus on dropping 0.6 support at this point. See #2218.

**Review by lefou — COMMENTED** [review]
Looks good. Only one question about visibility.

File: scalajslib/src/mill/scalajslib/api/ScalaJSApi.scala (hunk 1 above)
Comment (lefou) [reply scalajslib/src/mill/scalajslib/api/ScalaJSApi.scala:116]: As the `apply` and the `withX` methods are public, there is no reason to keep these `val`s non-public.

File: scalajslib/worker/0.6/src/ScalaJSWorkerImpl.scala (hunk 2 above)
Comment (lolgab) [reply scalajslib/worker/0.6/src/ScalaJSWorkerImpl.scala:87]: Yes, this is meant to go very soon, so we don't want to invest time on it :)

File: scalajslib/src/mill/scalajslib/api/ScalaJSApi.scala (hunk 1 above)
Comment (lolgab) [reply scalajslib/src/mill/scalajslib/api/ScalaJSApi.scala:116]: Inspired by Scala.js. This configuration class is meant to be a "write-only", there is no usage on the reading side, that's why I think they make the values package-private [here](https://github.com/scala-js/scala-js/blob/1ec64710273cec8ba41da4aa4c94dfa97d3fb872/linker-interface/shared/src/main/scala/org/scalajs/linker/interface/OutputPatterns.scala#L31).

Comment (lolgab) [reply scalajslib/src/mill/scalajslib/api/ScalaJSApi.scala:116]: The `apply` is private as well

Comment (lefou) [reply scalajslib/src/mill/scalajslib/api/ScalaJSApi.scala:116]: I just noticed the `apply` is private, too. Still, I think, if we can create it with some method e.g. `withJSFile` accepting a `String`, than there is sooner or later someone who will want to programmatically retrieve that `String`, e.g. to manipulate it slightly. 

If there is no other good reason (as we don't protect use from extra API maintenance here), we should not make it harder that necessary and make it public.

Comment (lolgab) [reply scalajslib/src/mill/scalajslib/api/ScalaJSApi.scala]: Usually, I prefer to follow whatever the Scala.js plugin does, but what you said makes sense.
Made them public ✅

**Review by lefou — APPROVED** [review]

**Pull request merged**